### ISSUE TITLE
NSWindow.IsVisible binding merges two difference definitions

### ISSUE THREAD

**Issue by Therzok** [xamarin/xamarin-macios#14359]
### Steps to Reproduce

1. Use NSWindow.IsVisible.
2. Observe it's not really what you expected because `setIsVisible` sets apple script metadata, and orderOut/orderFront should be used.

### Expected Behavior
`[NSWindow isVisible]` refers to the window being on screen as is defined on NSWindow
`[NSWindow setIsVisible:]` refers to the value that is shown to Apple Scripting, and comes from the NSWindowScripting protocol.

The bindings would have generated:
```csharp
public bool IsVisible { get; }

// or whatever is the general pattern for protocol implemented members.
void INSWindowScripting.SetIsVisible(bool value) {}
```

### Actual Behavior

```csharp
public bool IsVisible { get; set; }
```

### Environment
6.0.200 dotnet

**Comment by Therzok** [comment]
More info on this, as I have read up on the topic, it seems that IsVisible is synchronous and messes with NSWindow ref-counting. That means there is no clean way to ensure the NSWindow is released when using this API.

NSWindow's `makeKeyAndOrderFront:`, `orderFront:`, `orderOut:` handle the logic - including `setLevel:`, which would need to be done paired with `setIsVisible:` for appkit window management.

It seems that `setIsVisible:` is synchronous, and that also causes behavioural difference, so the `IsVisible` setter can't just be migrated to the `order*` versions.

**Comment by rolfbjarne** [comment]
I've proposed a fix: #14390, I've obsoleted the IsVisible setter with an explanation, added a SetIsVisible method, and I did the same for the two other properties in the same boat (IsZoomed and IsMiniaturized).
